Prose/ProseLib/Sources/EditProfileFeature/AuthenticationView.swift

```diff
@@ -0,0 +1,161 @@
+//
+// This file is part of prose-app-macos.
+// Copyright (c) 2022 Prose Foundation
+//
+
+import AppLocalization
+import Assets
+import ComposableArchitecture
+import ProseUI
+import SwiftUI
+
+private let l10n = L10n.EditProfile.Authentication.self
+
+// MARK: - View
+
+struct AuthenticationView: View {
+  typealias ViewState = AuthenticationState
+  typealias ViewAction = AuthenticationAction
+
+  let store: Store<ViewState, ViewAction>
+
+  var body: some View {
+    ScrollView(.vertical, showsIndicators: false) {
+      WithViewStore(self.store) { viewStore in
+        VStack(spacing: 24) {
+          ContentSection(
+            header: l10n.PasswordSection.Header.label,
+            footer: l10n.PasswordSection.Footer.label
+          ) {
+            VStack(alignment: .leading) {
+              ThreeColumns(l10n.Password.Header.label) {
+                Button { viewStore.send(.changePasswordTapped) } label: {
+                  Text(verbatim: l10n.Password.ChangePasswordAction.label)
+                    .frame(maxWidth: .infinity)
+                }
+                .controlSize(.large)
+              }
+              SecondaryRow(l10n.RecoveryEmail.Header.label) {
+                Text(verbatim: viewStore.recoveryEmail)
+                Button(l10n.RecoveryEmail.EditAction.label) {
+                  viewStore.send(.editRecoveryEmailTapped)
+                }
+                .controlSize(.small)
+              }
+            }
+            .padding(.horizontal)
+          }
+          Divider()
+            .padding(.horizontal)
+          ContentSection(
+            header: l10n.MfaSection.Header.label,
+            footer: l10n.MfaSection.Footer.label
+          ) {
+            VStack(alignment: .leading) {
+              ThreeColumns(l10n.MfaToken.Header.label) {
+                Button { viewStore.send(.disableMFATapped) } label: {
+                  Text(verbatim: l10n.MfaToken.DisableMfaAction.label)
+                    .frame(maxWidth: .infinity)
+                }
+                .controlSize(.large)
+              }
+              SecondaryRow(l10n.MfaStatus.Header.label) {
+                HStack(spacing: 4) {
+                  // TODO: Change this to a more general purpose indicator
```

Comment:
I mean, we have indicator dots all around the UI. This one has no specific meaning, so it could be refactored out so we can use it in other places.

I already created `ConnectionStatusIndicator` and `OnlineStatusIndicator`, maybe one could be renamed or a third one could be created.

Reply:
`ConnectionStatus` is `(disconnected|connected)`, but maybe we could rename it and change this to `(on|off)` or something.

Reply:
I don't know. The TODO just isn't very specific and I'm generally not a big fan to treat code like a ticketing system.